### PR TITLE
changes to file exports

### PR DESCRIPTION
**Pull request by lentschi**
Hi folks!

Two bugfixes:
- #524 affects CSV, Excel and HTML export
- #525 affects CSV export

Implemented feature request #526 (s. issue description)

I used `PHPExcel` as a git submodule, so if you want to try it out, you will have to run `git submodule update --init`.

As I wrote in #526: Should you consider merging, I'm open for any suggestions to round things up. (For now I only programmed this, to get a time sheet export in the way one customer of mine requires them.)

Cheers, Flo

### CODE REVIEW

File: core/extensions/ki_export/processor.php [hunk 1]
```diff
@@ -276,24 +276,16 @@
...
-          'decimal_separator' => $_REQUEST['decimal_separator'],
           'reverse_order' => isset($_REQUEST['reverse_order'])?1:0),
           'ki_export.xls.');      
        
         $exportData = export_get_data($in,$out,$filterUsers,$filterCustomers,$filterProjects,$filterActivities,false,$reverse_order,$default_location,$filter_cleared,$filter_type,false,$filter_refundable);
-        for ($i=0;$i<count($exportData);$i++) {
-          $exportData[$i]['decimalDuration'] = str_replace(".",$_REQUEST['decimal_separator'],$exportData[$i]['decimalDuration']);
-          $exportData[$i]['rate'] = str_replace(".",$_REQUEST['decimal_separator'],$exportData[$i]['rate']);
-          $exportData[$i]['wage'] = str_replace(".",$_REQUEST['decimal_separator'],$exportData[$i]['wage']);
-        }
```
Comment (simonschaufi) [inline core/extensions/ki_export/processor.php:288]: why did you remove this?

File: core/extensions/ki_export/processor.php [hunk 2]
```diff
@@ -276,24 +276,16 @@
...
         $view->exportData = count($exportData)>0?$exportData:0;
 
         $view->columns = $columns;
         $view->custom_timeformat = $timeformat;
         $view->custom_dateformat = $dateformat;
 
-        header("Content-Disposition:attachment;filename=export.xls");
-        header("Content-Type: application/vnd.ms-excel");
```
Comment (simonschaufi) [inline core/extensions/ki_export/processor.php]: why did you remove this?

File: core/extensions/ki_export/processor.php (hunk 1 above)
Comment (lentschi) [inline core/extensions/ki_export/processor.php:288]: @simonschaufi 
Because, when using Excel number formats instead of plain text (which was my main motivation for the commit - s. #526), one cannot set a **decimal seperator**, but [only a **locale**](https://github.com/lentschi/kimai/blob/699ff9b6f847e0855a1ec48d32b035d360f99747/core/extensions/ki_export/templates/scripts/formats/excel.php#L104), which I set to kimai config's `language` field for now. (So when using `en`, you should get a dot as a separator, with a `de` a comma...)

Like I mentioned in #526, should you consider merging this, I could add more **configuration options**. Since this is my first potential contribution to this project, I wasn't sure what the community would want to configure. Here are some suggestions though:
- Export language (For example, as a German speaker I might want my interface in German, but still export the data in English, since I have an English customer - that should also influence the decimal separator.)
- Turn headers/sums on/off.
- Modify date/time formats in the "floater" dialog (Currently the date format can only be modified via config's `date_format_1` and the time format is hardcoded to `%H:%M`)

Also, if people should want to still be able to export in the original excel format, I could add a `Format`-dropdown similar to the one in the pdf export floater, where users can decide about which basic template, they wanna use. (In that case I would re-add the decimal separator.)

**By the way - off topic:**

You appear to be one of the people, who contributed a lot to kimai - Thank you very much, it's a great tool - I use it a lot!

Comment (simonschaufi) [inline core/extensions/ki_export/processor.php:288]: ok thank you for your explanation!

**Comment by simonschaufi** [comment]
Thank you for your contribution! It would have been better to make multiple PRs, then I could merge them one by one. Next time...

**Comment by lentschi** [comment]
@simonschaufi 
Oh - I see...
Could I not still close this pull request and create three new ones (one for each commit)?

**Comment by simonschaufi** [comment]
I don't want to cause more trouble for you, so for now just leave it. It will test it asap

**Comment by lentschi** [comment]
I wouldn't be any trouble... But I don't want to close the PR if you're trying to merge it now, so I'll leave it open unless you tell me otherwise. ;-)
Also you could still cherry-pick:
http://mispdev.blogspot.co.at/2013/02/github-cherry-picking-commits-from-pull.html

**Comment by kevinpapst** [comment]
There seem to be merge conflicts @lentschi 
Could you please update your PR to the latest master?

**Comment by simonschaufi** [comment]
Don't worry, I can do that on the command line as well

**Comment by lentschi** [comment]
Okay - as Simon suggested I now closed the original PR and created two new ones (for the two minor bugfixes)
I didn't create one for the #526 feature just now, as it's buggy after resolving merge conflicts. (I believe due to [this change](https://github.com/kimai/kimai/commit/8ed1645d0017062e609effbaea32d64377b3ee45#diff-2080bcdbc95cd718934c1e9ff2f9401aR74)
However the merge is in the tracking branch at https://github.com/lentschi/kimai/ and apart from this problem, it seems to work.

**Comment by lentschi** [comment]
@simonschaufi 
I now get [your first question ](https://github.com/kimai/kimai/pull/527#discussion_r48302685) - I had it in the same commit as the bugfix - Sorry for the mess!
I now split it up and created yet another PR for the bugfix ;-)
